inskinesis/inskinesis_test.go

```diff
@@ -0,0 +1,205 @@
+			Records: []*kinesis.PutRecordsResultEntry{
+				{
+					ErrorCode: aws.String("error1"),
+				},
+				{
+					ErrorCode: aws.String("error2"),
+				},
+			},
+		}
+		records := []*kinesis.PutRecordsRequestEntry{
+			{
+				Data: []byte("record1\n"),
+			},
+			{
+				Data: []byte("record2\n"),
+			},
+		}
+		expected := [][]byte{[]byte("record1\n"), []byte("record2\n")}
+		actual := getFailedRecords(response, records)
+		assert.Equal(t, expected, actual)
+	})
+
+	t.Run("it_should_return_empty_slice_when_no_failed_records", func(t *testing.T) {
+		response := &kinesis.PutRecordsOutput{
+			Records: []*kinesis.PutRecordsResultEntry{
+				{
+					ErrorCode: nil,
+				},
+				{
+					ErrorCode: nil,
+				},
+			},
+		}
+		records := []*kinesis.PutRecordsRequestEntry{
+			{
+				Data: []byte("record1\n"),
+			},
+			{
+				Data: []byte("record2\n"),
+			},
+		}
+		expected := [][]byte{}
+		actual := getFailedRecords(response, records)
+		assert.Equal(t, expected, actual)
+	})
+}
+
+func Test_transformRecords(t *testing.T) {
+	s := stream{
+		partitioner: PartitionerPointer(Partitioners.UUID),
+	}
+
+	t.Run("it_should_return_transformed_records_correctly", func(t *testing.T) {
+		records := []interface{}{
+			map[string]string{
+				"key1": "value1",
+			},
+			map[string]string{
+				"key2": "value2",
+			},
+		}
+		expected := []*kinesis.PutRecordsRequestEntry{
+			{
+				Data: []byte("{\"key1\":\"value1\"}\n"),
+			},
+			{
+				Data: []byte("{\"key2\":\"value2\"}\n"),
+			},
+		}
+		actual, err := s.transformRecords(records)
+		assert.Nil(t, err)
+
+		assert.Equal(t, expected[0].Data, actual[0].Data)
+		assert.Equal(t, expected[1].Data, actual[1].Data)
+	})
+
+	t.Run("it_should_return_error_when_failed_to_transform_records", func(t *testing.T) {
+		records := []interface{}{make(chan int)}
+		transformed, err := s.transformRecords(records)
+		assert.Error(t, err)
+		assert.Nil(t, transformed)
+	})
+}
+
+func Test_CreateBatches(t *testing.T) {
+	t.Run("it_should_return_batches_correctly", func(t *testing.T) {
+		records := []interface{}{
+			map[string]string{
+				"key1": "value1",
+			},
+			map[string]string{
+				"key2": "value2",
+			},
+			map[string]string{
+				"key3": "value3",
+			},
+			map[string]string{
+				"key4": "value4",
+			},
+		}
+		expected := [][]interface{}{
+			{
+				map[string]string{
+					"key1": "value1",
+				},
+				map[string]string{
+					"key2": "value2",
+				},
+			},
+			{
+				map[string]string{
+					"key3": "value3",
+				},
+				map[string]string{
+					"key4": "value4",
+				},
+			},
+		}
+		actual, err := CreateBatches(records, 2, 100)
+		assert.Nil(t, err)
+		assert.Equal(t, expected, actual)
+	})
+
+	t.Run("it_should_return_error_when_failed_to_create_batches", func(t *testing.T) {
+		records := []interface{}{make(chan int)}
+		batches, err := CreateBatches(records, 2, 100)
+		assert.Error(t, err)
+		assert.Nil(t, batches)
+	})
+}
+
+//
+//func Test_putRecords(t *testing.T) {
```

Comment:
we should either delete or fix this test func

Reply:
fixed